pkg/asset/installconfig/aws/client.go

```diff
@@ -0,0 +1,117 @@
+package aws
+
+import (
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/route53"
+	"github.com/pkg/errors"
+)
+
+//go:generate mockgen -source=./client.go -destination=.mock/awsclient_generated.go -package=mock
+
+// API represents the calls made to the API.
+type API interface {
+	GetPublicHostedZones() (map[string][]string, error)
+	GetPublicHostedZone(zoneName string) (*route53.HostedZone, error)
+	GetPublicHostedZoneByID(zoneID string) (*route53.HostedZone, error)
+}
+
+// Client makes calls to the AWS API.
+type Client struct {
+	session *session.Session
+}
+
+// NewClient initializes a client with a session
+func NewClient() (*Client, error) {
+	return &Client{session: nil}, nil
+}
+
+// getSession will set the session if it's null. This is a hack to allow gomock
+// to run unit tests without the need for a real session.
+func (c *Client) getSession() (*session.Session, error) {
+	if c.session == nil {
+		session, err := GetSession()
+		if err != nil {
+			return nil, err
+		}
+
+		c.session = session
+	}
+	return c.session, nil
+}
```

Comment:
creating a new session for the client means that you can't use the session created by the install-config which includes the setup for AWS service endpoint overrides.

So let's make sure we never create a new session, and always re-use the sesssion from installconfigAsset.AWS.GetSession()

Reply:
Fixed!